core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -14,34 +14,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- package kafka.utils
+package kafka.utils
 
-import kafka.common._
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import kafka.utils.json.JsonValue
 import scala.collection._
-import util.parsing.json.JSON
 
 /**
  *  A wrapper that synchronizes JSON in scala, which is not threadsafe.
  */
-object Json extends Logging {
-  val myConversionFunc = {input : String => input.toInt}
-  JSON.globalNumberParser = myConversionFunc
-  val lock = new Object
+object Json {
+
+  private val mapper = new ObjectMapper()
 
   /**
    * Parse a JSON string into an object
    */
-  def parseFull(input: String): Option[Any] = {
-    lock synchronized {
-      try {
-        JSON.parseFull(input)
-      } catch {
-        case t: Throwable =>
-          throw new KafkaException("Can't parse json string: %s".format(input), t)
-      }
-    }
-  }
-  
+  def parseFull(input: String): Option[JsonValue] =
+    try Option(mapper.readTree(input)).map(JsonValue(_))
+    catch { case e: JsonProcessingException => None }
```

Comment:
Thanks for the review @rajinisivaram. The behaviour is the same. The catch block in the old code doesn't seem to do anything. For example:

```scala
scala> JSON.parseFull("}")
res1: Option[Any] = None
```